.github/workflows/docs-check.yml

```diff
@@ -45,5 +45,9 @@ jobs:
       - name: Install mkdocs and dependencies
         run: cd build/release/ && make deps.docs
 
+      - name: Build documentation for CRDs
```

Comment:
This `docs-check` action is only triggered in PRs. With the current approach, I don't believe it will be triggered to generate the doc when the docs are actually published by the `build/release/Makefile`. Instead of just generating the doc during the build, what about the following approach?
1. `make build` triggers the generation of the doc
2. The user would commit the generated specification.md with their other changes
3. The `docs-check` action would generate the docs and then validate that there are no changes in the doc. If there are changes, then fail the CI. The user would need to fix the committed file.
4. No special step needed during the docs publish, since the specification.md is already committed.

This approach allows us to see the diff in the specification.md between commits and branches

Reply:
Make sense the change I'm working on this